# Diff contra bitnami/charts@keycloak/25.2.0 (commit d7ec0ea7bff7264df7068ec6e51fbd3493d2c890)
# Comparacion: https://github.com/bitnami/charts/tree/keycloak/25.2.0/bitnami/keycloak
#               contra vendor/keycloak/ en este repo (Chart.yaml, values.yaml, templates/).
# No incluye charts/postgresql ni charts/common (dependencias OCI declaradas, no copias
# vendorizadas modificadas por EDNEL-RIOJA).

--- bitnami-keycloak-upstream/bitnami/keycloak/Chart.yaml	2026-10-06 13:46:22.570867877 +0200
+++ wk-dist-docs/vendor/keycloak/Chart.yaml	2026-10-06 13:41:19.989234491 +0200
@@ -1,13 +1,10 @@
-# Copyright Broadcom, Inc. All Rights Reserved.
-# SPDX-License-Identifier: APACHE-2.0
-
 annotations:
   category: DeveloperTools
   images: |
     - name: keycloak
-      image: docker.io/bitnami/keycloak:26.3.3-debian-12-r0
+      image: quay.io/keycloak/keycloak:26.3.2
     - name: keycloak-config-cli
-      image: docker.io/bitnami/keycloak-config-cli:6.4.0-debian-12-r11
+      image: docker.io/adorsys/keycloak-config-cli:6.4.0-26
   licenses: Apache-2.0
   tanzuCategory: application
 apiVersion: v2
@@ -36,4 +33,4 @@
 name: keycloak
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/keycloak
-version: 25.2.0
\ No newline at end of file
+version: 25.2.0

--- bitnami-keycloak-upstream/bitnami/keycloak/values.yaml	2026-10-06 13:46:22.571646581 +0200
+++ wk-dist-docs/vendor/keycloak/values.yaml	2026-10-06 13:41:20.002196940 +0200
@@ -91,9 +91,16 @@
 ## @param image.debug Enable Keycloak image debug mode
 ##
 image:
+  # La imagen de Bitnami (en bitnamilegacy desde que Bitnami retiro su catalogo publico). Las
+  # plantillas son las del chart de Bitnami: los initContainers cargan /opt/bitnami/scripts y
+  # copian /opt/bitnami/keycloak/*, que en la imagen oficial de quay.io/keycloak no existen
+  # (prepare-write-dirs falla con "liblog.sh: No such file or directory").
   registry: docker.io
-  repository: bitnami/keycloak
-  tag: 26.3.3-debian-12-r0
+  repository: bitnamilegacy/keycloak
+  # -debian-12-r2 y no "26.3.2": ese alias es la build r0 (jul 2025), cuyo entrypoint espera la
+  # BD en KEYCLOAK_DATABASE_HOST (defecto "postgresql") e ignora KC_DB_URL. La r2 es la misma
+  # que corre on-prem (harbor c1-apps/keycloak, digest 7043f871...).
+  tag: "26.3.2-debian-12-r2"
   digest: ""
   ## Specify a imagePullPolicy
   ## ref: https://kubernetes.io/docs/concepts/containers/images/#pre-pulled-images
@@ -1002,8 +1009,8 @@
   ##
   image:
     registry: docker.io
-    repository: bitnami/keycloak-config-cli
-    tag: 6.4.0-debian-12-r11
+    repository: adorsys/keycloak-config-cli
+    tag: 6.4.0-26
     digest: ""
     ## Specify a imagePullPolicy
     ## ref: https://kubernetes.io/docs/concepts/containers/images/#pre-pulled-images
@@ -1292,4 +1299,50 @@
   existingSecretUserKey: ""
   existingSecretPasswordKey: ""
   annotations: {}
-  extraParams: ""
\ No newline at end of file
+  extraParams: ""
+
+## EDNEL custom: post-import realm hardening Job (plantilla templates/realm-hardening-job.yaml)
+realmHardening:
+  enabled: false
+  image: quay.io/keycloak/keycloak:26.3.2
+  imagePullPolicy: IfNotPresent
+  backoffLimit: 5
+  ttlSecondsAfterFinished: 600
+  waitRetries: 60
+  waitSleepSeconds: 5
+  keycloakUrl: ""
+  adminUser: "admin"
+  adminPassword: ""
+  realm: ""
+  # frontendUrl del realm master. Vacio (default) = Keycloak usa KC_HOSTNAME, o sea
+  # el host publico del participante. Se rellena solo con adminRealmAccess=internal,
+  # para que la consola de administracion se AUTENTIQUE contra el host interno y no
+  # contra el publico: sin esto la consola se sirve en el host interno pero pide el
+  # login al publico, y ademas no se podria cerrar /realms/master/* en internet.
+  masterFrontendUrl: ""
+  ssoIdleTimeoutSeconds: 1800
+  masterPasswordPolicy: "length(16) and passwordHistory(20) and notUsername(undefined) and notEmail(undefined) and upperCase(1) and lowerCase(1) and digits(1) and maxLength(32) and forceExpiredPasswordChange(90)"
+  nonMasterPasswordPolicy: "length(16) and passwordHistory(20) and notUsername(undefined) and notEmail(undefined) and upperCase(1) and lowerCase(1) and digits(1) and specialChars(1) and maxLength(32) and forceExpiredPasswordChange(90)"
+  vault:
+    addr: ""
+    role: ""
+    # Secret (clave ca.crt) con la CA del endpoint de OpenBao. Vacio = CA publica.
+    tlsSecret: ""
+    envImage: "ghcr.io/bank-vaults/vault-env:v1.22.1"
+# Capa 1 del aislamiento del realm master: cierra /auth/admin/* y
+# /auth/realms/master/* en el host PUBLICO del participante. Apagado por
+# defecto -> los participantes externos no se ven afectados. Lo enciende el
+# deployer con adminRealmAccess=internal.
+masterRealmPublicBlock:
+  enabled: false
+  # Host publico donde se cierra (participant-be-<ns>.<dominio>). Obligatorio
+  # si enabled=true.
+  host: ""
+  # Tiene que ganar a la ruta que sirve /auth (un Ingress -> priority 0).
+  priority: 100
+  statusCode: 404
+  paths:
+    - /auth/realms/master
+    - /auth/realms/master/*
+    - /auth/admin
+    - /auth/admin/*

Only in wk-dist-docs/vendor/keycloak/templates: master-realm-block-route.yaml
Only in wk-dist-docs/vendor/keycloak/templates: realm-hardening-job.yaml
